rustls/src/quic.rs

```diff
@@ -763,6 +763,12 @@ impl<'a> KeyBuilder<'a> {
     }
 }
 
+/// A provider for QUIC initial keys.
+pub trait KeyProvider: Debug + Send + Sync {
```

Comment:
So QUIC has a fixed cipher suite for the initial connection key set. If we want a consistent `CryptoProvider` for the cipher suite implementation stored, it seems optimal if the `CryptoProvider` can generate the initial keys (with the well-known `Tls13CipherSuite` that provider uses for `TLS13_AES_128_GCM_SHA256`). Unfortunately it feels like we've painted ourselves into a corner, because we can't add fields to `CryptoProvider` without breaking semver compatibility and that seems to be the only way we can extend `CryptoProvider` functionality.

I guess QUIC implementations will have to pass in the correct `Tls13CipherSuite` explicitly and will be required to check at run-time that `Tls13CipherSuite::quic` is not `None`.

Reply:
Could we add something like this? It wouldn't necessarily have to live on `CryptoProvider`, it could also live in `rustls::quic` as a free function.

```rust
impl CryptoProvider {
    pub fn quic_initial_key_set_cipher_suite(
        &self,
    ) -> Option<&'static crate::tls13::Tls13CipherSuite> {
        self.cipher_suites
            .iter()
            .find(|s| s.suite() == crate::CipherSuite::TLS13_AES_128_GCM_SHA256)
            .and_then(|found| found.tls13())
            .filter(|found| found.quic.is_some())
    }
}
```

Reply:
Hmm, seems reasonable, I'll play with that.

Reply:
Worked out a slightly different take, please have a look.